pallets/automation-time/src/tests.rs

```diff
@@ -280,29 +355,234 @@ fn schedule_transfer_with_dynamic_dispatch() {
+		let expected_schedule_fee_amount =
+			calculate_local_action_schedule_fee(weight, num_of_execution);
+
+		assert_eq!(fee_amount, expected_schedule_fee_amount);
+	})
+}
+
+#[test]
+fn calculate_dynamic_dispatch_action_schedule_fee_amount_works() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let num_of_execution = 2;
+		let call: <Test as frame_system::Config>::RuntimeCall =
+			frame_system::Call::remark_with_event { remark: vec![0] }.into();
+		let action = Action::DynamicDispatch { encoded_call: call.encode() };
+
+		let fee_amount = AutomationTime::calculate_schedule_fee_amount(&action, num_of_execution)
+			.expect(EXPECT_CALCULATE_SCHEDULE_FEE_AMOUNT);
+
+		let weight = <Test as Config>::WeightInfo::run_dynamic_dispatch_action()
+			.saturating_add(call.get_dispatch_info().weight);
 		let expected_schedule_fee_amount =
-			FEE_PER_SECOND * XCMP_TASK_REF_TIME * (execution_count as u128) /
-				(WEIGHT_REF_TIME_PER_SECOND as u128);
-		assert_eq!(schedule_fee_amount, expected_schedule_fee_amount);
+			calculate_local_action_schedule_fee(weight, num_of_execution);
+
+		assert_eq!(fee_amount, expected_schedule_fee_amount);
+	})
+}
+
+#[test]
+fn calculate_xcmp_action_schedule_fee_amount_works() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let num_of_execution = 2;
+		let action = create_xcmp_task_action(XcmpTaskParams::default());
+		let fee_amount = AutomationTime::calculate_schedule_fee_amount(&action, num_of_execution)
+			.expect(EXPECT_CALCULATE_SCHEDULE_FEE_AMOUNT);
+		let expected_schedule_fee_amount = calculate_expected_xcmp_task_schedule_fee(
+			action.schedule_fee_location::<Test>(),
+			num_of_execution,
+		);
+		assert_eq!(fee_amount, expected_schedule_fee_amount);
+	})
+}
+
+#[test]
+fn calculate_xcmp_action_schedule_fee_amount_with_different_schedule_fees_works() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		ASSET_FEE_PER_SECOND.into_iter().for_each(|fee| {
+			let num_of_execution = 2;
+			let action = create_xcmp_task_action(XcmpTaskParams {
+				schedule_fee: fee.asset_location.clone(),
+				..XcmpTaskParams::default()
+			});
+
+			let fee_amount =
+				AutomationTime::calculate_schedule_fee_amount(&action, num_of_execution)
+					.expect(EXPECT_CALCULATE_SCHEDULE_FEE_AMOUNT);
+
+			let expected_schedule_fee_amount =
+				calculate_expected_xcmp_task_schedule_fee(fee.asset_location, num_of_execution);
+
+			assert_eq!(fee_amount, expected_schedule_fee_amount);
+		});
+	})
+}
+
+#[test]
+fn calculate_xcmp_action_schedule_fee_amount_with_absolute_or_relative_native_schedule_fee_works() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let num_of_execution = 2;
+
+		let action = create_xcmp_task_action(XcmpTaskParams {
+			schedule_fee: MultiLocation::new(
+				1,
+				X1(Parachain(<Test as Config>::SelfParaId::get().into())),
+			),
+			..XcmpTaskParams::default()
+		});
+		let fee_amount_1 = AutomationTime::calculate_schedule_fee_amount(&action, num_of_execution)
```

Comment:
Please use `fee_amount_abosolute` instead of `fee_amount_1`. The same goes to all `_1` and `_2`s. What are 1 and 2?